pkg/controller/s3/bucket/taggingConfig.go

```diff
@@ -61,13 +60,30 @@ func (in *TaggingConfigurationClient) Observe(ctx context.Context, bucket *v1bet
 		return Updated, nil
 	case config == nil && len(external.TagSet) != 0:
 		return NeedsDeletion, nil
-	case cmp.Equal(s3.SortS3TagSet(external.TagSet), s3.SortS3TagSet(GenerateTagging(config).TagSet)):
+	case IsS3TagsUpdated(s3.SortS3TagSet(external.TagSet), s3.SortS3TagSet(GenerateTagging(config).TagSet)):
 		return Updated, nil
 	default:
 		return NeedsUpdate, nil
 	}
 }
 
+func IsS3TagsUpdated(externalTagSets []awss3.Tag, generatedTagSet []awss3.Tag) bool {
```

Comment:
This function compares whether generated tags are a subset of external tags. So I think it's better to rename the function, for example `subtags`.

Reply:
good point